bgpd/bgp_vty.c

```diff
@@ -6712,6 +6734,22 @@ bgp_show_summary_afi_safi (struct vty *vty, struct bgp *bgp, int afi, int safi,
         {
           if (is_wildcard)
             {
+              /*
+               * So limit output to those afi/safi pairs that
+               * actualy have something interesting in them
+               */
+              int count = bgp_show_summary_afi_safi_peer_count (bgp, afi, safi);
+              if (!count)
+                {
+                  safi++;
+                  if (safi == SAFI_RESERVED_4 ||
```

Comment:
If you move this special case check to bgp_show_summary_afi_safi_peer_count it only needs to be in once place (second place is below.)